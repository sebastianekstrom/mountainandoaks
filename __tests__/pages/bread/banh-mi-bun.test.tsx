import React from "react";
import { render, screen } from "@testing-library/react";
import BanhMiBun from "pages/bread/banh-mi-bun";

describe("BanhMiBun page", () => {
  it("renders title correctly", () => {
    render(<BanhMiBun />);
    const title = screen.queryAllByText(/Bánh mì bun/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<BanhMiBun />);
    const ingress = screen.queryAllByText(
      /This iconic Vietnamese bread is known/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the ingredients", () => {
    render(<BanhMiBun />);
    const waterLabel = screen.queryAllByText(/Water/i);
    const eggLabel = screen.queryAllByText(/Egg/i);
    expect(waterLabel).toBeDefined();
    expect(eggLabel).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<BanhMiBun />);
    const step = screen.queryAllByText(
      /Start by melting the butter over low heat/i
    );
    expect(step).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<BanhMiBun />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });
});
