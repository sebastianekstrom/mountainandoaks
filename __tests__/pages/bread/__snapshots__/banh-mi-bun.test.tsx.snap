// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BanhMiBun page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Bread flour (12% protein)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          500
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Water
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          260
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class="flex flex-row items-center"
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Egg
          </p>
          <span
            class="ml-2 flex items-center"
          >
            <button
              class="outline-none focus:shadow-[0_0_0_2px] focus:shadow-black"
              data-state="closed"
            >
              <img
                alt=""
                class="w-[15] h-auto"
                height="15"
                quality="100"
                src="/images/misc/question-mark.svg"
                width="15"
              />
            </button>
          </span>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          60
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Butter
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          30
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          9
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Yeast
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          7
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Dough conditioner (optional)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sugar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          6
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class="flex flex-row items-center"
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Melted butter
          </p>
          <span
            class="ml-2 flex items-center"
          >
            <button
              class="outline-none focus:shadow-[0_0_0_2px] focus:shadow-black"
              data-state="closed"
            >
              <img
                alt=""
                class="w-[15] h-auto"
                height="15"
                quality="100"
                src="/images/misc/question-mark.svg"
                width="15"
              />
            </button>
          </span>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        />
      </li>
    </ul>
    <div
      class="ml-auto"
    >
      <button
        class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
        data-testid="BAKERS_PERCENTAGE"
        role="button"
      >
        <p
          class="variable font-sans text-caption leading-paragraph "
        >
          Show bakers %
        </p>
      </button>
    </div>
  </div>
</section>
`;
